client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -8,10 +9,120 @@
     remove_container_metadata
 )
-from ayon_core.hosts.substancepainter.api.lib import prompt_new_file_with_mesh
 
 import substance_painter.project
-import qargparse
+
+
+def _convert(subst_attr):
+    """Function to convert substance C++ objects to python instance.
+    It is made to avoid any possible ValueError when C++ objects casting
+    as python instance.
+
+    Args:
+        subst_attr (str): Substance attributes
+
+    Raises:
+        ValueError: Raise Error when unsupported Substance
+            Project was detected
+
+    Returns:
+        python instance: converted python instance of the C++ objects.
+    """
+    if subst_attr in {"Default", "UVTile", "TextureSetPerUVTile"}:
+        return getattr(substance_painter.project.ProjectWorkflow, subst_attr)
+    elif subst_attr in {"PerFragment", "PerVertex"}:
+        return getattr(substance_painter.project.TangentSpace, subst_attr)
+    elif subst_attr in {"DirectX", "OpenGL"}:
+        return getattr(substance_painter.project.NormalMapFormat, subst_attr)
+    else:
+        raise ValueError(
+            f"Unsupported Substance Objects: {subst_attr}")
+
+
+def parse_substance_attributes_setting(template_name, project_templates):
+    """Function to parse the dictionary from the AYON setting to be used
+    as the attributes for Substance Project Creation
+
+    Args:
+        template_name (str): name of the template from the setting
+        project_templates (dict): project template data from the setting
+
+    Returns:
+        dict: data to be used as attributes for Substance Project Creation
+    """
+    attributes_data = {}
+    for template in project_templates:
+        if template["name"] == template_name:
+            attributes_data.update(template)
+    attributes_data["normal_map_format"] = _convert(
+        attributes_data["normal_map_format"])
+    attributes_data["project_workflow"] = _convert(
+        attributes_data["project_workflow"])
+    attributes_data["tangent_space_mode"] = _convert(
+        attributes_data["tangent_space_mode"])
+    attributes_data.pop("name")
+    attributes_data.pop("preserve_strokes")
+    return attributes_data
+
+
+def parse_subst_attrs_reloading_mesh(template_name, project_templates):
+    """Function to parse the substances attributes ('import_cameras'
+        and 'preserve_strokes') for reloading mesh
+        with the existing projects.
+
+    Args:
+        template_name (str): name of the template from the setting
+        project_templates (dict): project template data from the setting
+
+    Returns:
+        dict: data to be used as attributes for reloading mesh with the
+            existing project
+    """
+    attributes_data = {}
+    for template in project_templates:
+        if template["name"] == template_name:
+            for key, value in template.items():
+                if isinstance(value, bool):
+                    attributes_data.update({key: value})
+    return attributes_data
```

Comment:
I really have a hard time understanding the differences between these functions.
These are much easier if they just take a template - and are explicit about what keys they take instead of checking on boolean types, etc.
And then having a dedicated 'get_template' function or whatever if really needed.

For example, it's also unclear which of these are now setting the "Default Texture Resolution" value from settings?

```suggestion
def get_project_settings(template):
    """Return project settings based on template from settings.

    Args:
        template (dict): template from settings

    Returns:
        dict: data to be used as attributes for Substance Project Creation
    """
    keys = [
        "normal_map_format",
        "project_workflow",
        "tangent_space_mode",
        "default_texture_resolution"
    ]
    return {
        key: _convert(template[key]) for key in keys
    }

def get_mesh_reloading_settings(template):
    """Function to parse the substances attributes ('import_cameras'
        and 'preserve_strokes') for reloading mesh
        with the existing projects.

    Args:
        template (dict): template from settings

    Returns:
        dict: data to be used as attributes for reloading mesh with the
            existing project
    """
    keys = ["import_cameras", "preserve_strokes"]
    return {
       key: template.get(key, False) for key in keys
    }
```

Note that this first one will only work if `_convert` ignores the integer value from `default_texture_resolution` key.
But - this may be even further simplified if we just explicitly create the `substance_painter.project.Settings` so we don't need these functions at all:

```python
settings = substance_painter.project.Settings(
        normal_map_format=_convert(template["normal_map_format"]),
        project_workflow=_convert(template["project_workflow"]),
        tangent_space_mode=_convert(template["tangent_space_mode"]),
        default_texture_resolution=template["default_texture_resolution"]
)`
This seems quite explicit, quite readable and very clear what is expected from the template - instead of having separate functions that also appear rather complex?

Reply:
But does it work when there are multiple templates existing?